### PR TITLE
UT Spatial Structure 6

### LINKED ISSUE #64: Checks throwing confusing error
the Ifc checking results are throwing an error I do not understand what is wrong with it. This relates to checking of SpatialStructure-6-3-RC2.ifc in pull request #62

```
<MVDMessage>
      <MessageType>ERROR</MessageType>
      <MVDType>SCHEMADEFINITION</MVDType>
      <MessageCode>SYNTAX_VALIDATION_1</MessageCode>
      <Instance>#46596=IFCTRANSPORTELEMENTTYPE('0a5MZ59PD0wwcMc0ilz4oQ',&#36;,'RHD_GM_Vessel:RHD_GM_Vessel',&#36;,&#36;,&#40;#46593,#46590&#41;,&#40;#46598&#41;,'339908',&#36;,VEHICLEMARINE)</Instance>
      <InstanceLink></InstanceLink>
      <Messages>
        <MessageString> incompatible operand types 'PredefinedType' = '_enum_literal'&#10;First Operand: OID: #46596&#10;Type: IfcTransportElementType&#10;References: SELFVEHICLEMARINE&#10;Type: IfcTransportElementNonFixedTypeEnum&#10;Second Operand: USERDEFINED&#10;Type: IfcTransportElementFixedTypeEnum&#10;in file ..&#92;ifc_schemata&#92;ifc4x3_rc2.exp at line 11842&#10;</MessageString>
      </Messages>
</MVDMessage>
```

I think there is an error in the tool as I see nothing wrong with the syntax...

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
## Unit test

This scenario establishes a spatial structure for a shipyard. The intention is to show and test how spatial structures can be used to organize a project. The model utilizes new vertical and regional breakdown objects to organize the project and provide a skeleton for local placement.

The IFC files anre encoded RC3 so will need to be checked once checking tool is ready.
## Covered topics

*tick one/many*

- [x] project setup
- [ ] georeferencing
- [ ] alignment semantics
- [ ] alignment geometry
- [ ] linear placement
- [x] project breakdown structure / spatial structure
- [ ] terrain
- [ ] triangulated irregular network
- [ ] open cross profile
- [ ] sectioned surface & solid
- [ ] pavement & course
- [ ] surface feature
- [ ] earthworks
- [ ] geotechnics
- [ ] longitudinal products (barriers, guardrail, ...)
- [ ] interchange
- [ ] structural
- [ ] drainage
- [ ] signage
- [ ] road furniture
- [ ] rail furniture
- [ ] ports & waterways furniture
- [ ] bridge furniture

### CODE REVIEW

**Comment by jmirtsch** [comment]
I'd suggest the transport elements should be contained in the space, not aggregating as per this image.

210309 element containment

**Comment by AlexBrad1eyCT** [comment]
@jmirtsch thanks for catching that the inevitable overlook of code reuse! Fixed in latest commit

**Comment by pjanck** [comment]
Can you please merge the `main` branch again and rerun the tests? The checker just got updated.

**Comment by pjanck** [comment]
How is error-fixing progressing? Having any difficulties?

File: SpatialStructure-6/readme.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+				- `Representation=IfcBoundingBox`
+				- `PredefinedType=IfcSpaceTypeEnum.BERTH`
+		- S.SY.L.4 `IfcMarineFacility` (Dry Dock 1)
+			- `Representation=IfcBoundingBox`
+			- `PredefinedType=IfcMarineFacilityTypeEnum.DRYDOCK`
+
+
+The application should be able to display a project hierarchy as shown in Figure 2.
+
+![Project Tree][figure 3]
```
Comment (pjanck) [inline SpatialStructure-6/readme.md]: Do you mean figure 3 here?

Comment (AlexBrad1eyCT) [reply SpatialStructure-6/readme.md]: Yep good spot, correction in next commit

**Comment by AlexBrad1eyCT** [comment]
I'm sorting out some style issues in the revit export im extracting elements from, revit is exporting `IfcPresentationStyleAssignment` objects which are deprecated(causing most of the errors I think). And I couldn't find the delete for GGIfc (jon has told me now). This should sort that out. Will push when I update my cleanup method on my scripts.

**Comment by jmirtsch** [comment]
> 
> 
> I'm sorting out some style issues in the revit export im extracting elements from, revit is exporting `IfcPresentationStyleAssignment` objects which are deprecated(causing most of the errors I think). And I couldn't find the delete for GGIfc (jon has told me now). This should sort that out. Will push when I update my cleanup method on my scripts.

I'll email you, I'll help make the toolkit upgrade the presentationstyleassignment rather than delete it.

**Comment by AlexBrad1eyCT** [comment]
Solved all errors, that relate to file, only thing left is a variation of the KIT problem with Where rules and on IfcElementType subtypes

**Comment by AlexBrad1eyCT** [comment]
@pjanck  Sucess! PDT where rules all functioning! i will now merge!

**Pull request merged**